sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceScanExecRedactionSuite.scala

```diff
@@ -244,4 +244,33 @@ class DataSourceV2ScanExecRedactionSuite extends DataSourceScanRedactionTest {
       }
     }
   }
+
+  test("SPARK-37585: test input metrics for DSV2 with output limits") {
```

Comment:
I should have pointed it out earlier... It's weird to put metrics test in this suite, how about `FileBasedDataSourceSuite`?

Reply:
Sure, but should I move the other related tests together in this PR as well?

Reply:
don't you have only one new test? any other related tests?

Reply:
I meant the other test related to InputMetrics here.

Reply:
yea let's move them together

Reply:
oh i see. yea, okay to move that together.